easybuild/easyblocks/m/mcr.py

```diff
@@ -0,0 +1,146 @@
+##
+# Copyright 2009-2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for installing MCR, implemented as an easyblock
+
+@author: Stijn De Weirdt (Ghent University)
+@author: Dries Verdegem (Ghent University)
+@author: Kenneth Hoste (Ghent University)
+@author: Pieter De Baets (Ghent University)
+@author: Jens Timmerman (Ghent University)
+@author: Fotis Georgatos (Uni.Lu, NTUA)
+"""
+
+import re
+import os
+import shutil
+
+from distutils.version import LooseVersion
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_MCR(EasyBlock):
+    """Support for installing MCR."""
+
+    def __init__(self, *args, **kwargs):
+        """Add extra config options specific to MCR."""
+        super(EB_MCR, self).__init__(*args, **kwargs)
+        self.comp_fam = None
+        self.configfilename = "my_installer_input.txt"
+        self.extradir =  {'R2014a': 'v83', 'R2014b': 'v84', 'R2015a': 'v85'}
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'java_options': ['-Xmx256m', "$_JAVA_OPTIONS value set for install and in module file.", CUSTOM],
+        }
+        return EasyBlock.extra_options(extra_vars)
+
+    def configure_step(self):
+        """Configure MCR installation: create license file."""
+
+        configfile = os.path.join(self.builddir, self.configfilename)
+        try:
+            if LooseVersion(self.version) < LooseVersion('2015a'):
+                shutil.copyfile("%s/installer_input.txt" % self.builddir, configfile)
+                config = file(configfile).read()
+                regdest = re.compile(r"^# destinationFolder=.*", re.M)
+                regagree = re.compile(r"^# agreeToLicense=.*", re.M)
+                regmode = re.compile(r"^# mode=.*", re.M)
```

Comment:
no need to compile these separately, since they're used only once:

```python
config = re.sub(r"^# destinationFolder=.*", "destinationFolder=%s" % self.installdir, config, re.M)
...
```